lib/motsu/src/context.rs

```diff
@@ -1,34 +1,97 @@
 //! Unit-testing context for Stylus contracts.
-use std::sync::{Mutex, MutexGuard};
 
+use std::{collections::HashMap, ptr};
+
+use dashmap::DashMap;
+use once_cell::sync::Lazy;
 use stylus_sdk::{alloy_primitives::uint, prelude::StorageType};
 
-use crate::storage::reset_storage;
+use crate::prelude::{Bytes32, WORD_BYTES};
 
-/// A global static mutex.
-///
-/// We use this for scenarios where concurrent mutation of storage is wanted.
-/// For example, when a test harness is running, this ensures each test
-/// accesses storage in an non-overlapping manner.
+/// Context of stylus unit tests associated with the current test thread.
+#[allow(clippy::module_name_repetitions)]
+pub struct TestContext {
```

Comment:
In nearest future it is possible to add to this `TestContext` struct fiels as `msg_sender`, `msg_value` and so on.